test/gtest/ucp/test_ucp_tag_match.cc

```diff
@@ -748,6 +748,7 @@ UCS_TEST_P(test_ucp_tag_match_rndv, exp_huge_mix) {
         std::vector<char> recvbuf(size, 0);
 
         ucs::fill_random(sendbuf);
+        VALGRIND_MAKE_MEM_UNDEFINED(&recvbuf[0], recvbuf.size());
```

Comment:
why is it added only here? should we add it in the rx req init flow?

Reply:
it's because std::vector constructor makes the memory defined. would not happen if we allocated the receive buffer by malloc/mmap, for example.

Reply:
is this the only place where we use vector for rx buffer?

Reply:
there are many tests that use std::vector, I guess many would have to be re-written to catch (potentially) undefined memory

Reply:
@brminich is it ok to merge w/o changing all tests, since some fixes in this PR are needed to pass Valgrind with MLNX_OFED 5.4?